core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -236,8 +254,8 @@ public long nextFetchOffset() {
                 NavigableMap<Long, InFlightBatch> subMap = cachedState.subMap(baseOffset, true, endOffset, true);
                 if (subMap.isEmpty()) {
                     // The in-flight batches are removed, might be due to the start offset move. Hence,
-                    // the next fetch offset should be the partition end offset + 1.
-                    nextFetchOffset = endOffset + 1;
+                    // the next fetch offset should be the partition end offset.
+                    nextFetchOffset = endOffset;
```

Comment:
This shouldn't change, else we ll get last batch in fetch response for replica manager.

Reply:
Change seems good.